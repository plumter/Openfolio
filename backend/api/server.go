package api

import (
	"fmt"

	"github.com/gin-gonic/gin"
	"github.com/plumter/Openfolio/api/routers"
	"github.com/plumter/Openfolio/pkg"
	"github.com/plumter/Openfolio/repository/models"
	"github.com/plumter/Openfolio/token"
)

type Server struct {
	router *gin.Engine
}

func NewServer(config pkg.Configurations, service models.Service) (*gin.Engine, error) {
	tokenMaker, err := token.NewPasetoMaker(config.TokenSymmetricKey)

	if err != nil {
		return nil, fmt.Errorf("cannot create token maker: %e ", err)
	}
	router := gin.New()
	router.Use(gin.Logger())
	router.Use(gin.Recovery())

	organizationConfig := &routers.UserRouteConfig{
		TokenMaker: tokenMaker,
		Service:    service,
		Config:     &config,
		Router:     router,
	}
	routers.UserRoutes(organizationConfig)

	return router, nil
}
