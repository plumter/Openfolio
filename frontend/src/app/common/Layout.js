
const Layout = ({children}) => {
    
    return  <div className="flex justify-center" data-testid="current-page">
                <div className="layout">
                    <div>
                        {children}
                    </div>
                </div>
            </div>
}

export default Layout;