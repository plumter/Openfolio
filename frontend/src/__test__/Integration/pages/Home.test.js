import "../common/server";
import {waitFor, screen, act} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { logoutUser, startApp, waitForToastMessage } from "../common/base";

// Start App
beforeEach(async () => {
  await startApp("/",  BrowserRouter);
});

describe("Home Page Test", _ => {
  it('Test Home Page View', async () => {
    expect(screen.getByTestId("current-page")).toMatchSnapshot();
  });

  it('Test Fail Sign-in Link Request', async () => {
    // Input Email and password then Submit
    userEvent.type(screen.getByPlaceholderText('example@mail.com'), 'eni@plumter.com',);
    // Wait for validation debounce
    await act(() => new Promise(r => setTimeout(r, 80)));
    userEvent.click(screen.getByText("Send Sign-In Link"));

    // Wait For Error message
    await waitForToastMessage("Invalid Request");
  });

  it('Test Successful Sign-in Link Request', async () => {
    // Input Email and password then Submit
    userEvent.type(screen.getByPlaceholderText('example@mail.com'), 'hammed@plumter.com',);
    // Wait for validation debounce
    await act(() => new Promise(r => setTimeout(r, 80)));
    userEvent.click(screen.getByText("Send Sign-In Link"));

    // Wait For Success View
    await act(() => new Promise(r => setTimeout(r, 100)));
    await waitFor(() => screen.getByText("Sign-In Link Sent"));
    expect(screen.getByTestId("current-page")).toMatchSnapshot();
  });

});