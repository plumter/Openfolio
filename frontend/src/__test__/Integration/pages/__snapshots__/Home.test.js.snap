// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Test Test Home Page View 1`] = `
<div
  class="flex justify-center"
  data-testid="current-page"
>
  <div
    class="layout"
  >
    <div>
      <div
        class="my-auto pt-24 sm:w-[33rem] mx-auto px-7 sm:px-10"
      >
        <form>
          <div
            class="lg:px-5"
          >
            <div
              class="flex pb-8"
            >
              <a
                class="flex-none my-auto"
                href="/"
              >
                <img
                  alt="P"
                  class="w-6"
                  src="/assets/media/logos/logo.png"
                />
              </a>
              <h2
                class="pl-2 my-auto text-lg"
              >
                openfolio
              </h2>
            </div>
            <div
              class="pb-2"
            >
              <h6
                class="font-semibold text-lg "
              >
                Enter your Email
              </h6>
              <p
                class="text-grey-2 py-1.5"
              >
                Enter your email to receive a link to create, edit or manage your profile.
              </p>
              <input
                class="form-input"
                name="email"
                pattern="[a-zA-Z0-9\\\\._%!$&*=^|~#%'\`?{}/\\\\-]+@([a-zA-Z0-9\\\\-]+\\\\.){1,}([a-zA-Z]{2,16})"
                placeholder="example@mail.com"
                required=""
                type="email"
              />
              <div
                style="height: 0px; visibility: hidden;"
              >
                <div
                  class="react-reveal text-xs mb-1 pt-2 text-red-500"
                  style="opacity: 0;"
                >
                  Enter a valid email address
                </div>
              </div>
            </div>
            <div
              class="mt-4 mb-7"
            >
              <button
                class="w-full btn-primary"
                disabled=""
                type="submit"
              >
                Send Sign-In Link
              </button>
            </div>
          </div>
        </form>
      </div>
      <footer
        class="mt-auto pt-12"
      >
        <div
          class="flex justify-center"
        >
          <p
            class="font-medium pr-2 my-auto"
          >
            Powered By:
          </p>
          <a
            href="https://plumter.com"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="P"
              class="w-7"
              src="/assets/media/logos/plumter-logo.png"
            />
          </a>
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`Home Page Test Test Successful Sign-in Link Request 1`] = `
<div
  class="flex justify-center"
  data-testid="current-page"
>
  <div
    class="layout"
  >
    <div>
      <div
        class="my-auto pt-24 sm:w-[33rem] mx-auto px-7 sm:px-10"
      >
        <div
          class="mb-5 text-center"
        />
        <h6
          class="flex-1 font-semibold text-lg text-center py-1.5"
        >
          Sign-In Link Sent
        </h6>
        <p
          class="lg:px-8 text-center pb-8 text-grey-2 text-base"
        >
          We've sent an email with a sign-in link to 
          <span
            class="text-primary"
          >
             
            hammed@plumter.com
          </span>
          . Check your inbox and click on the link provided to sign into Openfolio.
        </p>
      </div>
      <footer
        class="mt-auto pt-12"
      >
        <div
          class="flex justify-center"
        >
          <p
            class="font-medium pr-2 my-auto"
          >
            Powered By:
          </p>
          <a
            href="https://plumter.com"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="P"
              class="w-7"
              src="/assets/media/logos/plumter-logo.png"
            />
          </a>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
